metadata/src/main/java/com/redhat/lightblue/metadata/FieldConstraint.java

```diff
@@ -31,4 +31,10 @@
      * Determines if the constraint is valid for the given field type
      */
     boolean isValidForFieldType(Type fieldType);
+
+
+    /**
+     * Returns the description of the FieldConstraint (for documentation propose)
+     */
+    String getDescription();
```

Comment:
Maybe I chose the wrong word for the method name, it basically helps to generate an user text for a constraint and it can vary a lot. After the iterations we had for the issue that this PR aism to solve, the regex constraint (which uses 'pattern') is the only one using it right now and on this case it use to describe the pattern used for that constraint (which can vary for each request)